File: api/src/main/java/fr/huiitre/tools/modules/riot/valorant/application/usecase/GetValorantVersionUseCase.java
package fr.huiitre.tools.modules.riot.valorant.application.usecase;

import java.util.Map;
import java.util.Optional;

import org.springframework.stereotype.Service;

import fr.huiitre.tools.modules.core.module.domain.ModuleCode;
import fr.huiitre.tools.modules.core.role.domain.RoleCode;
import fr.huiitre.tools.modules.core.security.application.usecase.SecuredUseCase;
import fr.huiitre.tools.modules.riot.valorant.application.ports.ValorantVersionProvider;

@Service
public class GetValorantVersionUseCase implements SecuredUseCase {

    private final ValorantVersionProvider versionProvider;

    public GetValorantVersionUseCase(ValorantVersionProvider versionProvider) {
        this.versionProvider = versionProvider;
    }

    @Override
    public Optional<ModuleCode> requiredModule() {
        return Optional.of(ModuleCode.RIOT);
    }

    @Override
    public RoleCode requiredRole() {
        return RoleCode.READ_ONLY;
    }

    public Map<String, Object> execute() {
        return versionProvider.getVersion();
    }
}

File: api/Modules/Riot/Valorant/Application/Core/Usecases/GetValorantVersionUseCase.cs
using Tools.Api.Modules.Core.Security.Application.Services;
using Tools.Api.Modules.Core.Security.Application.Usecases;
using Tools.Api.Modules.Core.Security.Domain;
using Tools.Api.Modules.Riot.Valorant.Application.Core.Ports;

namespace Tools.Api.Modules.Riot.Valorant.Application.Core.Usecases;

public sealed class GetValorantVersionUseCase(
    UseCaseAuthorizer authorizer,
    IValorantVersionProvider versionProvider
) : SecuredUseCase(authorizer)
{
    protected override RoleCode RequiredRole => RoleCode.ReadOnly;
    protected override ModuleCode? RequiredModule => ModuleCode.Riot;

    public Task<IReadOnlyDictionary<string, object>> Execute()
    {
        return versionProvider.GetVersion();
    }
}
